Development/cpprest/json_utils.cpp

```diff
@@ -74,17 +75,25 @@ namespace web
         // returns true if the value has at least one field matching the key path
         // if any arrays are encountered on the key path, results is an array, otherwise it's a non-array value
         bool extract(const web::json::object& object, web::json::value& results, const utility::string_t& key_path)
+        {
+            std::vector<utility::string_t> key_path_;
+            boost::algorithm::split(key_path_, key_path, [](utility::char_t c) { return U('.') == c; });
+
+            return extract(object, results, key_path_);
+        }
+
+        // find the value of a field or fields from the specified object and searching key path arrays as necessary
+        // returns true if the value has at least one field matching the key path
+        // if any arrays are encountered on the key path, results is an array, otherwise it's a non-array value
+        bool extract(const web::json::object& object, web::json::value& results, const std::vector<utility::string_t>& key_path)
         {
             bool match = false;
             results = web::json::value::null();
 
             std::list<const web::json::object*> pobjects(1, &object);
-            utility::string_t::size_type key_first = 0;
-            do
+            for (auto key : key_path)
             {
-                const utility::string_t::size_type key_last = key_path.find_first_of(_XPLATSTR("."), key_first);
-                const utility::string_t key = key_path.substr(key_first, details::count(key_first, key_last));
-                if (utility::string_t::npos != key_last)
+                if (key != *(key_path.end() - 1))
```

Comment:
That checks that the _value_ of the key isn't equal to the last one in the `key_path`... but what about paths like `foo.bar.foo`?

Reply:
Well-spotted will fix it